### ISSUE TITLE
Invalid date error

### ISSUE THREAD

**Issue by efox29** [ethereum/mist#525]
Node type: geth/eth
OS: Windows 10 64 bit
Network type: main
Mist version: 0.7.2

I participated in the DOA Creation yesterday and was transferring ETH around between accounts, and after I was done (transferring, watch contracts and watch tokens), I have been getting the same error and the program is very slow.  Seems like something I did, created this problem. 

Looking at my processes, geth.exe takes up about 750MB of ram, and there is disk activity (seems like something is happening).

I have checked my date, and the time and date is correct.
I have also downloaded the entire blockchain again.

Error:
![invalid](https://cloud.githubusercontent.com/assets/304290/14942996/e7e1f03a-0f9b-11e6-87ba-f84d7a486a28.png)

Console:
![console](https://cloud.githubusercontent.com/assets/304290/14942999/06a64a02-0f9c-11e6-9cf1-997bfc919801.png)

**Comment by Film54** [comment]
I am having the exact same problem...  and I have transactions that are stuck in this void.

![invalid date](https://cloud.githubusercontent.com/assets/17620974/14943198/72bb6fca-0f87-11e6-9075-ae0a35f9bcca.jpg)

**Comment by super83** [comment]
I'm having the same problem with my wallet too.
It all started yesterday after i send Ether to DAO and added "Watch Token" and "Watch Contract" in my wallet....and now i starting receiving this error, "invalid date" on top of my wallet !!!

Please somebody fix this problem soon....thanks

**Comment by Film54** [comment]
Out of the blue and on its own, my issue has been resolved...

**Comment by hawaii1234** [comment]
When my wallet open , I see the "Invalid date" error. stopped downloading more blocks...  How to fix it ?

**Comment by clund** [comment]
Having similar issue on MacOSX. I hope I don't have to dl the entire chain again (for the nth time).

![screen shot 2016-05-02 at 19 46 41](https://cloud.githubusercontent.com/assets/1596463/14962332/a3f5fc8c-109e-11e6-88e8-c8f9869c62d4.png)

**Comment by blockq4** [comment]
Same issue started after setting my watch contract for DAO. Using 0.7.2 - Windows 10 x64

**Comment by Marty45** [comment]
Same issue,

**Comment by Film54** [comment]
My issue is back.  It started working again randomly yesterday for about 30 minutes... now it's back to "invalid date".

**Comment by speedy029** [comment]
same issue here, deleting contract helps but that's not the solution

**Comment by CaseyWA** [comment]
Running 0.7.2 on Win 10 64bit, same issue as described above started today.  I've been watching the DAO contract and DAO token since it launched with no issues prior to 5/3/2016.  

Running geth outside of Mist to download the blockchain prior to even installing Mist due to some mining.  Again, haven't seen any issues prior to today when I unlocked my screen.

Closed Mist, re-opened, let run for an hour with no change.  Have not attempted to reboot computer yet as it's busy running some jobs.

**Comment by 0ldPaul** [comment]
Same Issue, invalid date running 0.7.2 on Win 10 64bit, date sync with internet date but nothing happened. Like that since 5/3/2016..

**Comment by clund** [comment]
Rebooted. Opened MIST, got the "invalid date" message, but after  minute or so the sync started. MacOSX, 0.7.2

**Comment by luclu** [comment]
Can confirm:
<img width="421" alt="unbenannt-1" src="https://cloud.githubusercontent.com/assets/10135872/15012424/d60ee016-11f8-11e6-8c83-840f1fb19d3d.png">
This happens on startup for several seconds.

**Comment by blockq4** [comment]
Devs any progress on this? My balances are correct but I see no transactions into wallet, Invalid Date - have tried all ideas so far and nothing works - Windows 10 x64 0.7.2

**Comment by luclu** [comment]
@blockq4 this is currently high priority.

**Comment by jordimorris** [comment]
How are we going to recommend Mist to our friends if there's an error right in our faces?

**Comment by frozeman** [comment]
@karalabe @bas-vk @fjl  its either the IPC connection dropping (@efox29), which could be because the node crashes, or it doesn't sync on windows (@Film54), which can be firewall issues.

Its working flawless on mac, tho.

@jordimorris i understand, we are working on it, but building such a complex system which runs on multiple platforms is not an easy one. Best is you don't recommend it for now, we have more traction than we can handle right now anyway :)

**Comment by SWKStudios** [comment]
@frozeman I only have the issue (windows 10 64bit) when I added watch contract for TheDAO. This is the only contract I have ever added. I removed it and now works fine again. I still have TheDAO added to "Watch Token" without an issue.

@jordimorris Isn't everything semi in beta still? It sucks I admit, but right now I expect some issues. We can still use command line via geth if need be.

**Comment by hisnamesollie** [comment]
@frozeman I'm having the same issue on Mac. Tried installing to an older version (which has worked for others), but still the same issue.

**Comment by hisnamesollie** [comment]
I've been running 0.7.2 and 0.6.1

**Comment by alexvandesande** [comment]
I've been investigating this and it seems to be a corrupted database file. If this affected you, try deleting (or just renaming) the folder /library/Ethereum/chaindata 

Unfortunately it means you'll have to resend it, yes. We are still investigating the cause.

**Comment by efox29** [comment]
@alexvandesande I have already deleted the chaindata folder and redownloaded the blockchain again. I still have the same issue. I followed the instructions here 
http://ethereum.stackexchange.com/questions/3529/how-can-i-force-a-redownload-of-the-entire-blockchain-on-the-mist-wallet

**Comment by taylormiles** [comment]
Same issue.   Mac OSX 10.11.4  Mist version 0.7.2.   Issue did start shortly after I followed the daohub.org Token and Contract.  I will try and remove these.

**Comment by rideSpecialized** [comment]
Just to add a data point.  Running Win7Ent x64.  0.7.2 

Wallet has been running flawlessly right up until I added TheDAO contract and watched TheDAO token.

I'm showing connected to between 8 and 12 peers.  Showing "0" (zero) blocks.  And the date flaps between the clock icon and "Invalid Date".  I have not tried to send/receive any transactions since this began.

Ethereum-Wallet

TheDAO token is disappeared from the interface.  TheDAO contract is still there (and very very slow) and I'm receiving the following status message when attempting to delete it:

"Do you really want to delete this wallet contract? The account will still exist on the blockchain but 
if you delete the address, you will not be able to find and will lose access to the funds deposited."

Unclear from that status message if I will lose access to the funds "in the wallet" or if I will lose my access to those tokens permanently (I assume it's just the wallet but don't want to chance it).

**Comment by Kozzi07** [comment]
I also had a lot of problem by watching DAO contract in Mist 0.7.2. (Win 64bit)
Mist slow/freezing when clicking on contract and sometimes not downloading block for a while. (need to be restarted)
Back on 0.6.2 and no more issue so there must be something wrong in 0.7.2 (perhaps linked to contract events?)

**Comment by SWKStudios** [comment]
@rideSpecialized you can delete "The DAO Contact" from your watched contracts and then re-add it without any issues. I have done it a couple of times now.

**Comment by rideSpecialized** [comment]
Follow up datapoint... I removed TheDAO contract from my client and restarted the client.  100% back to normal instantly.  @SWKStudios thanks for the help... I will keep my eyes open for an update on the issue.

**Comment by efox29** [comment]
@rideSpecialized is this the workaround ? Did you add the contract back and was it still a problem ?

**Comment by rideSpecialized** [comment]
@efox29 I did not try to re-add the contract yet.  But removing it seemed to fix the problem instantly (after restarting the client, as a precaution).

**Comment by CaseyWA** [comment]
Based on some of the other comments, it appears this issue is related to adding the DAO contract to your wallet.  In case additional information is needed to determine what about having the contract in your wallet causes this issue, here is another data point based on what I observed this evening, not sure if this will help in troubleshooting the issue or not but thought I'd capture the information:

Using the "developer tools" -> "wallet ui" view I was looking around and saw some failures generated from files archived within the *.asar files.  I attempted to rename both to .old to see how app would respond (no surprise, it failed to launch), then started to download and install node.js when I noticed that suddenly my wallet was synced back up without the "Invalid Date" error.  Cancelled out of the node.js install.  Block data showed synced to the current block and kept in step (have geth running in a cmd window 24/7).

I closed the Mist wallet, re-opened and the "Invalid date" error is back.  Here is what I see in the developer view in terms of errors, unsure if they are related to the actual problem or not:

Failed to load resource: net::ERR_FILE_NOT_FOUND
file:///C:/Users/%USERNAME%/Desktop/Ethereum-Wallet-win64-0-7-2/resources/app.as…/wallet/packages/ethereum_dapp-styles/fonts/SourceSansPro-ExtraLightIt.otf Failed to load resource: net::ERR_FILE_NOT_FOUND
085e9d7….js:303 Connect to node...

.....

CONNECTION ERROR: Couldn't connect to node on IPC.
    at Object.module.exports.InvalidConnection (C:\Users\%USERNAME%\Desktop\Ethereum-Wallet-win64-0-7-2\resources\app.asar\node_modules\web3\lib\web3\errors.js:28:16)
    at IpcProvider._timeout (C:\Users\%USERNAME%\Desktop\Ethereum-Wallet-win64-0-7-2\resources\app.asar\node_modules\web3\lib\web3\ipcprovider.js:150:48)
    at C:\Users\%USERNAME%\Desktop\Ethereum-Wallet-win64-0-7-2\resources\app.asar\node_modules\web3\lib\web3\ipcprovider.js:42:15
    at EventEmitter.<anonymous> (C:\Users\%USERNAME%\Desktop\Ethereum-Wallet-win64-0-7-2\resources\app.asar\modules\ipc\ipcProviderWrapper.js:51:13)
    at emitTwo (events.js:87:13)
    at EventEmitter.emit (events.js:172:7)


I've had the wallet open for another 20 minutes or so and the error persists.  Attempted to repeat what was done initially but no change in the "Invalid date" error.  I'm not sure what might have caused it to temporarily start working (maybe just some sort of timing issue) and since the short time it started working wasn't persistent, it doesn't appear to be related to anything I changed.

I'm glad to see there seems to be a workaround but would be great to find the actual fix.  For now I'll leave my wallet as is and if someone has a potential fix they want tested or would like some additional info on the issue, please let me know and I'll be happy to help wherever I can.

**Comment by hisnamesollie** [comment]
I've deleted the DAO contract and the wallet appears to be working like normal. Echoing above

**Comment by jordimorris** [comment]
Confirmed: the best option so far seems to just run 6.2 beta 13, works smooth also the dao contract interface that used to freeze in 7.2. 
@frozeman sorry but I'll recommend my friends to install 6.2, this is so BIG 
thank you for the tips and answers

**Comment by actuallymentor** [comment]
Same issue on 0.7.1 and 0.7.2. Mac OSX, invalid date since last block error in GUI.

**Comment by KentBarr** [comment]
I also have INVALID DATE

With 0.7.2 on Windows 7 laptop.

Previously it was working and I was able to get DAO tokens and all seemed fine. The problem appeared the next time I started etherem-wallet.

**Comment by housemobile** [comment]
Same error. Windows 7 with 0.7.2 and DAO watch token

**Comment by housemobile** [comment]
I can't even remove the watch token contract because Mist wallet freezes within seconds of opening it.

**Comment by rideSpecialized** [comment]
@housemobile it was extremely slow for me as well, but was not actually frozen (maybe 5+minutes between clicks... but it would work).  Eventually I was able to open the smart contract and hover over it.  Hovering will reveal trash can icon.  Click it, wait to execute, exit, and restart Mist.

**Comment by alexvandesande** [comment]
Yes it seems to be connected to adding the DAO contract too. Maybe the huge amount of events that it generated somehow screwed the local database? I recommend anyone who is having this issue to delete the dao contract from the watch list (you can still send ether to it via the send tab) and redownload the whole chain.

**Comment by mtbitcoin** [comment]
I was able to reproduce the error both with 1.3.6 and 1.4.1RC geth. Adding the DAO contract causes Mist to freeze. Removing the contract brings it bck to normal. At first i also thought it might have had something to do with the huge amount of events. But when I used a previous version of Mist i.e 0.5.2 the same contract, it would load up fine with no "freezes". 

Also, for those who are unable to DELETE the contract from mist 0.7.2 I would suggest using mist 0.5.2 to DELETE the contract and then switch back to 0.7.2

**Comment by dhelma** [comment]
Had same issue as well. My solution: Delete your Mist folder in User/*/AppData/Roaming to remove your contract watching, delete the chaindata in Ethereum if you want to start fresh. Now using 0.6.2, watching Dao and all is well so far.

**Comment by bardacuda** [comment]
I was getting the same invalid date error and freezing in 0.7.2 and it only started after I started watching the DAO contract. I am on Windows 7 64 bit and I made sure to sync my system clock with the NIST time server every time before I opened the wallet. The wallet would sync when I opened it and I would be connected to 5 - 15+ peers, but all transaction history would be blurred out and show as having 0 of 12 confirms (even though they had 10s, 100s, etc. confirms on the blockchain) and the wallet would be extremely slow/unresponsive.

Without deleting the watched contract or the blockchain data, I downloaded and ran 0.6.2 as was suggested. It synced and is running fine. I can see the DAO contract and my transaction history as normal. Whatever was causing this seems to be related to the newer Mist version and not to geth, corrupted blockchain data, or time sync issues.

**Comment by Beervangeer** [comment]
having the same problem. will go back to 0.6.2 to.

**Comment by callmetau** [comment]
Same issue. Mac OS. Tried to redownload blockchain - doesn't help. Fix it please ASAP.

**Comment by TrevorHilder** [comment]
Yes, I found my Ethereum Wallet 0.7.2 on Ubuntu 14.04 LTS in this state, having bought into The DAO last Saturday. I think the Wallet stated it had synced last block 11 hours ago when I looked at about 11:00 BST today. My attempts to get it going again failed, so I deleted the contents of the chaindata directory, and it took about two and a half hours to sync, then came up with "invalid date since last block". I have attached the log file to this comment. I deleted the Watch on The DAO contract, closed and re-opened the Wallet, and it came up normally.
[-1462547070019.txt](https://github.com/ethereum/mist/files/252552/-1462547070019.txt)

**Comment by KentBarr** [comment]
Confirming that on (Windows 7) switching to 0.6.2 worked to sync up and delete the watch contract. 

Then I quit and restarted with 0.7.2 and it worked. There's no invalid date error now, and no need to delete and download the blockchain.

FWIW the watch contract seemed to be working fine under 0.6.2

**Comment by derEremit** [comment]
confirming problems on Linux 64bit with 0.7.2
working fine by just launching 0.6.2

**Comment by goodc0re** [comment]
OSX 10.10.5 
Mist 0.7.2 
watching the DAO and the Tokens
also getting the "Invalid date since last block"

the message disappears after waiting a minute or two 
once it did not, and restarting Mist worked

**Comment by glesaint** [comment]
@KentBarr Thanks! Switching from 0.7.2 to 0.6.2 also solved this issue on osx 10.11.4. No need to delete chaindata or stop watching contracts.

**Comment by callmetau** [comment]
@glesaint I have Mac OS 10.11.4 and still have Invalid date issue 
Mist 0.7.2

**Comment by glesaint** [comment]
@callmetau can you try replacing Mist by its version 0.6.2? https://github.com/ethereum/mist/releases/tag/0.6.2

**Comment by callmetau** [comment]
@glesaint I've deleted The DAO contract. It fixed this issue

**Comment by msam4444** [comment]
Same error: Windows 10 with 0.7.2 and DAO watch token

**Comment by fineartcommodities** [comment]
invalid date on mist 0.7.2, win 10, after DAO watch. Resolved by restarting mist after switching date settings/automatic time zone settings...  More sessions after, the invalid date error yet reappear, just reswich timezone settings and it work....

**Comment by CryptoYeti** [comment]
Just adding my own data points.

Same error "Invalid Date" using Mist 0.7.2 on Windows 7 64 bit. Error appeared only AFTER adding the watch contract and watch token for the DAO. Reverted to earlier version (0.5.2) and error disappeared. Contract and token watch both seem to work for my needs, so I will await new release with bug fix.

I agree that there is no need to delete the chaindata file or stop watching contracts, simply downgrade to an earlier version until the bug is fixed.

**Comment by fineartcommodities** [comment]
win7 64 bit, mist 0.7.2 Print screen describing morning experience in sequence after some invalid date error resolved. (2 hours later all fine).

1) We can't estimate your gas usage properly, as you need at least 1 ether in your account (account was founded with 104.36... ether, sorry the img is not compleet)
![mist1](https://cloud.githubusercontent.com/assets/17668554/15111555/14ed1f08-15e8-11e6-9a85-b44d224b3307.png)

2) Connection error: couldn't connect to node on IPC
![mist3](https://cloud.githubusercontent.com/assets/17668554/15111563/1cfa907c-15e8-11e6-9666-e66a2da2c1f0.png)

3)couldn't connect to node did it crash in the background? 
![mist2](https://cloud.githubusercontent.com/assets/17668554/15111576/302339c4-15e8-11e6-9061-ecc5561f7934.png)

**Comment by tonefreqhz** [comment]
I have the same issue on MAC OSX 10.11.4 ElCapitan.
If I run a node mining in Console 
3) Start geth with --rpc option and also start up ethminer which will communicate with it over port 8545

Console 1:
geth --rpc

Console 2:
ethminer -G


and then open the mist wallet opens but does not keep up with synching to block chain although peers are attached?

**Comment by alexvandesande** [comment]
Please all update the wallet and try again.

**Comment by gitottof** [comment]
same issue ubuntu 15.04 
terminal gives the following warnings at startup:


[2016-05-13 13:05:22.443] [WARN] nodeConnector - NODECONNECTOR ERROR { [Error: connect ECONNREFUSED /home/gog/.ethereum/geth.ipc]
  code: 'ECONNREFUSED',
  errno: 'ECONNREFUSED',
  syscall: 'connect',
  address: '/home/gog/.ethereum/geth.ipc' }
Error: connect ECONNREFUSED /home/gog/.ethereum/geth.ipc
    at Object.exports._errnoException (util.js:856:11)
    at exports._exceptionWithHostPort (util.js:879:20)
    at PipeConnectWrap.afterConnect [as oncomplete] (net.js:1063:14)
[2016-05-13 13:05:22.647] [ERROR] main - Couldn't get time from NTP time sync server. [Error: can't find 'ntp enabled:' line in timedatectl output]
Error: can't find 'ntp enabled:' line in timedatectl output

**Comment by jordimorris** [comment]
Version 7.3 gives problems as well. it gets behind all the time. I can't believe this has been tested with 100 users and nobody had this issue.

6.2 is the best right now

**Comment by hisnamesollie** [comment]
@jordimorris I'm with you. As soon as I updated the wallet from 0.6.2 to 0.7.3 it was falling behind from the get go, without the DAO watch contract enabled. I've gone back to 0.6.2 without the DAO contract enabled for the time being, which works great.

**Comment by luclu** [comment]
This issue should be solved with geth 1.4.4 which comes bundled starting with 0.7.4.
Could you please update and verify?

**Comment by hisnamesollie** [comment]
Not for me. Did this a few days ago. Worked perfectly until I added the DAO watch contract, and the new wallet froze again like the others. I'm currently running 0.7.4 without the DAO contract.

**Comment by Kozzi07** [comment]
Same for me. Watching DAO contract made 0.7.4 freeze. (like other 0.7.x version) 
I am using it without contract but it will be a problem when we will have to interact with DAO contract...

**Comment by jordimorris** [comment]
I stick to version 6.2, it works smooth and no problems watching DAO contracts

**Comment by luclu** [comment]
Could you please try out the latest [release 0.7.5](https://github.com/ethereum/mist/releases)?

**Comment by CaseyWA** [comment]
Hi @luclu, I had a chance to try out release 0.7.5 tonight.  While still a bit slow syncing up with The DAO contract, it is actually functional now!  It also is displaying my token balance correctly which it was not before.  I see a "check for update" menu option which I'm hoping means the next update might be a simple pull through the GUI :-)  This was on a Win 10 machine, also tested it on a Ubuntu 15.10 box but did not have the DAO contract setup on that system.  Worked great and was fairly fast. 

I appreciate all the hard work that I'm sure went into this release and slowing things down a bit to make sure it was more stable then the last few.  

Thanks!!

**Comment by luclu** [comment]
I will close this specific thread as solved - as the initial issue does not seem to appear anymore and this gets cluttered.
Should you still encounter the "invalid date" message, please give some ping-back over here.
Otherwise feel free to open a new issue. 

---

That is good to hear @CaseyWA - looking forward to iron out the remaining big ones.
